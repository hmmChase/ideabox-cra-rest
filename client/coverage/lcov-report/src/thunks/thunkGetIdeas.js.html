<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/thunks/thunkGetIdeas.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\index.html">All files</a> / <a href="index.html">src/thunks</a> thunkGetIdeas.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>10/10</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>10/10</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import * as action from '../actions';
import * as api from '../api';
&nbsp;
export const getIdeas = url =&gt; {
  return async dispatch =&gt; {
    try {
      dispatch(action.hasErrored(false));
      dispatch(action.isLoading(true));
      const ideas = await api.request(url);
      dispatch(action.setIdeas(ideas));
      dispatch(action.isLoading(false));
    } catch (error) {
      dispatch(action.isLoading(false));
      dispatch(action.hasErrored(true));
    }
  };
};
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Dec 09 2018 06:33:58 GMT-0700 (Mountain Standard Time)
</div>
</div>
<script src="..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\sorter.js"></script>
</body>
</html>
