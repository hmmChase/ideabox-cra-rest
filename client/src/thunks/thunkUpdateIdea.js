import * as action from '../actions';
import * as api from '../api';

export const updateIdea = (ideaID, nextIdea, ideas) => {
  return async dispatch => {
    try {
      dispatch(action.hasErrored(false));
      await api.updateIdea(ideaID, nextIdea);
      const updatedIdeas = ideas.map(idea => {
        if (idea.id === ideaID) {
          return { ...idea, idea: nextIdea };
        } else {
          return idea;
        }
      });
      dispatch(action.setIdeas(updatedIdeas));
    } catch (error) {
      dispatch(action.hasErrored(true));
    }
  };
};
