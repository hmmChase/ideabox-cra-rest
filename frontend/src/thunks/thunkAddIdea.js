import * as action from '../actions';
import * as api from '../api';

export const addIdea = idea => async dispatch => {
  try {
    dispatch(action.hasErrored(false));
    const responseIdea = await api.postIdea(idea);
    dispatch(action.setIdea(responseIdea));
  } catch (error) {
    dispatch(action.hasErrored(true));
  }
};
