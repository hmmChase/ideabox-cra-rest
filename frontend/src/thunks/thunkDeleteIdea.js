import * as action from '../actions';
import * as api from '../api';

export const deleteIdea = (ideaID, ideas) => {
  return async dispatch => {
    try {
      dispatch(action.hasErrored(false));
      await api.deleteIdea(ideaID);
      const filteredIdeas = ideas.filter(idea => {
        return idea.id !== ideaID;
      });
      dispatch(action.setIdeas(filteredIdeas));
    } catch (error) {
      dispatch(action.hasErrored(true));
    }
  };
};
