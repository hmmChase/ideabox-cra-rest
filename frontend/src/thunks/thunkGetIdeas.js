import * as action from '../actions';
import * as api from '../api';

export const getIdeas = url => {
  return async dispatch => {
    try {
      dispatch(action.hasErrored(false));
      dispatch(action.isLoading(true));
      const ideas = await api.request(url);
      dispatch(action.setIdeas(ideas));
      dispatch(action.isLoading(false));
    } catch (error) {
      dispatch(action.isLoading(false));
      dispatch(action.hasErrored(true));
    }
  };
};
