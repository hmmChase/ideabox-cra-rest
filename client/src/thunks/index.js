import { getIdeas } from './thunkGetIdeas';
import { addIdea } from './thunkAddIdea';
import { updateIdea } from './thunkUpdateIdea';
import { deleteIdea } from './thunkDeleteIdea';

export { getIdeas, addIdea, updateIdea, deleteIdea };
