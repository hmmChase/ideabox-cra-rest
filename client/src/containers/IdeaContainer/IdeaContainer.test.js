import React from 'react';
import { shallow } from 'enzyme';
import {
  IdeaContainer,
  mapStateToProps,
  mapDispatchToProps
} from './IdeaContainer';
import * as thunk from '../../thunks';
jest.mock('../../thunks');

describe('IdeaContainer', () => {
  let wrapper;
  let mockProps;

  beforeEach(() => {
    jest.resetAllMocks();
    mockProps = {
      stateIsLoading: false,
      stateHasErrored: false,
      stateIdeas: [{ id: 1, idea: 'mock idea' }, { id: 2, idea: 'mock idea' }],
      dispatchGetIdeas: jest.fn()
    };

    wrapper = shallow(<IdeaContainer {...mockProps} />, {
      disableLifecycleMethods: true
    });
  });

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('matches the snapshot with props.stateisLoading: true', () => {
    mockProps.stateIsLoading = true;

    wrapper = shallow(<IdeaContainer {...mockProps} />, {
      disableLifecycleMethods: true
    });

    expect(wrapper).toMatchSnapshot();
  });

  it('matches the snapshot with props.stateHasErrored: true', () => {
    mockProps.stateHasErrored = true;

    wrapper = shallow(<IdeaContainer {...mockProps} />, {
      disableLifecycleMethods: true
    });

    expect(wrapper).toMatchSnapshot();
  });

  describe('componentDidMount', () => {
    it('calls getIdeas()', () => {
      const getIdeas = (wrapper.instance().getIdeas = jest.fn());

      wrapper.instance().componentDidMount();

      expect(getIdeas).toHaveBeenCalledTimes(1);
    });
  });

  describe('getIdeas', () => {
    it('calls dispatchGetIdeas(/api/v1/ideas/{id}) if props.id', () => {
      mockProps.id = 1;

      wrapper = shallow(<IdeaContainer {...mockProps} />, {
        disableLifecycleMethods: true
      });

      wrapper.instance().getIdeas();

      expect(mockProps.dispatchGetIdeas).toHaveBeenCalledTimes(1);
      expect(mockProps.dispatchGetIdeas).toHaveBeenCalledWith(
        `/api/v1/ideas/${mockProps.id}`
      );
    });

    it('calls dispatchGetIdeas(/api/v1/ideas) if no props.id', () => {
      mockProps.id = undefined;

      wrapper = shallow(<IdeaContainer {...mockProps} />, {
        disableLifecycleMethods: true
      });

      wrapper.instance().getIdeas();

      expect(mockProps.dispatchGetIdeas).toHaveBeenCalledTimes(1);
      expect(mockProps.dispatchGetIdeas).toHaveBeenCalledWith('/api/v1/ideas');
    });
  });

  describe('displayIdeaCards', () => {
    it('returns array with <IdeaCard> for each idea', () => {
      const ideaCards = wrapper.instance().displayIdeaCards();

      expect(Array(ideaCards)).toBeTruthy();
      expect(ideaCards).toHaveLength(2);
    });
  });

  describe('mapStateToProps', () => {
    it('parses state and maps to props', () => {
      const mockState = {
        isLoading: false,
        hasErrored: false,
        ideas: [{ id: 1, idea: 'mock idea' }]
      };

      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps.stateIsLoading).toEqual(mockState.isLoading);
      expect(mappedProps.stateHasErrored).toEqual(mockState.hasErrored);
      expect(mappedProps.stateIdeas).toEqual(mockState.ideas);
    });
  });

  describe('mapDispatchToProps', () => {
    it('calls dispatch on getIdeas(url)', () => {
      const mockDispatch = jest.fn();
      const mappedProps = mapDispatchToProps(mockDispatch);

      const mockUrl = '/api/vi/mock';

      mappedProps.dispatchGetIdeas(mockUrl);

      expect(mockDispatch).toHaveBeenCalledTimes(1);
      expect(thunk.getIdeas).toHaveBeenCalledTimes(1);
      expect(thunk.getIdeas).toHaveBeenCalledWith(mockUrl);
    });
  });
});
