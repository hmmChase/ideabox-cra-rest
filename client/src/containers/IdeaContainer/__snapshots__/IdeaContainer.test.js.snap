// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IdeaContainer matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <IdeaContainer
    dispatchGetIdeas={[MockFunction]}
    stateHasErrored={false}
    stateIdeas={
      Array [
        Object {
          "id": 1,
          "idea": "mock idea",
        },
        Object {
          "id": 2,
          "idea": "mock idea",
        },
      ]
    }
    stateIsLoading={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ul>
        <Connect(IdeaCard)
          id={2}
          idea="mock idea"
        />
        <Connect(IdeaCard)
          id={1}
          idea="mock idea"
        />
      </ul>,
      "className": "IdeaContainer",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(IdeaCard)
            id={2}
            idea="mock idea"
          />,
          <Connect(IdeaCard)
            id={1}
            idea="mock idea"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "ideaCard2",
          "nodeType": "class",
          "props": Object {
            "id": 2,
            "idea": "mock idea",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "ideaCard1",
          "nodeType": "class",
          "props": Object {
            "id": 1,
            "idea": "mock idea",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "ul",
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ul>
          <Connect(IdeaCard)
            id={2}
            idea="mock idea"
          />
          <Connect(IdeaCard)
            id={1}
            idea="mock idea"
          />
        </ul>,
        "className": "IdeaContainer",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Connect(IdeaCard)
              id={2}
              idea="mock idea"
            />,
            <Connect(IdeaCard)
              id={1}
              idea="mock idea"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "ideaCard2",
            "nodeType": "class",
            "props": Object {
              "id": 2,
              "idea": "mock idea",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "ideaCard1",
            "nodeType": "class",
            "props": Object {
              "id": 1,
              "idea": "mock idea",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "ul",
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;

exports[`IdeaContainer matches the snapshot with props.stateHasErrored: true 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <IdeaContainer
    dispatchGetIdeas={[MockFunction]}
    stateHasErrored={true}
    stateIdeas={
      Array [
        Object {
          "id": 1,
          "idea": "mock idea",
        },
        Object {
          "id": 2,
          "idea": "mock idea",
        },
      ]
    }
    stateIsLoading={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Error />,
      "className": "IdeaContainer",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Error />,
        "className": "IdeaContainer",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;

exports[`IdeaContainer matches the snapshot with props.stateisLoading: true 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <IdeaContainer
    dispatchGetIdeas={[MockFunction]}
    stateHasErrored={false}
    stateIdeas={
      Array [
        Object {
          "id": 1,
          "idea": "mock idea",
        },
        Object {
          "id": 2,
          "idea": "mock idea",
        },
      ]
    }
    stateIsLoading={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Loader />,
      "className": "IdeaContainer",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Loader />,
        "className": "IdeaContainer",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
